apollo-remote-state/client/src/operations/mutations/addTodo.tsx

```diff
@@ -0,0 +1,35 @@
+
+import { gql, useMutation } from "@apollo/client";
+import * as AddTodoTypes from './__generated__/AddTodo';
+import { GET_ALL_TODOS } from "../queries/getAllTodos";
+
+export const ADD_TODO = gql`
+  mutation AddTodo ($text: String!) {
+    addTodo (text: $text) {
+      success
+      todo {
+        id
+        text 
+        completed
+      }
+      error {
+        message
+      }
+    }
+  }
+`
+
+export function useAddTodo () {
+  const [mutate, { data, error }] = useMutation<
+    AddTodoTypes.AddTodo, 
+    AddTodoTypes.AddTodoVariables
+  >(
+    ADD_TODO,
+    {
+      refetchQueries: [{
```

Comment:
It should be possible to get away from using `refetchQueries` here (and everywhere else in these example apps). It would be great to show how the cache either updates automatically after a mutation, or how to specifically update the cache using `update` + `read/writeQuery`. We could also show `cache.modify` in advanced cases, if needed (but I think we're going to steer away from promoting `cache.modify` for anything other than advanced cases).